fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java

```diff
@@ -468,13 +477,23 @@ public void refreshTable(HiveMetaStoreTableInfo hmsTable)
 
     public void refreshPartition(HiveMetaStoreTableInfo hmsTable, List<String> partNames) throws DdlException {
         GlobalStateMgr.getCurrentState().getMetastoreEventsProcessor().getEventProcessorLock().writeLock().lock();
+        Map<HivePartitionKey, Future<HivePartition>> hivePartitionFutures = new HashMap();
         try {
             for (String partName : partNames) {
                 List<String> partValues = client.partitionNameToVals(partName);
                 HivePartitionKey key = new HivePartitionKey(hmsTable.getDb(), hmsTable.getTable(),
                         hmsTable.getTableType(), partValues);
-                partitionsCache.put(key, loadPartition(key));
-                partitionStatsCache.put(key, loadPartitionStats(key));
+                Future<HivePartition> partitionFuture = partitionRefreshExecutor.submit(() -> loadPartition(key));
+                hivePartitionFutures.put(key, partitionFuture);
+            }
+            for (Map.Entry<HivePartitionKey, Future<HivePartition>> entry : hivePartitionFutures.entrySet()) {
+                partitionsCache.put(entry.getKey(), entry.getValue().get());
+            }
+            for (String partName : partNames) {
+                List<String> partValues = client.partitionNameToVals(partName);
+                HivePartitionKey key = new HivePartitionKey(hmsTable.getDb(), hmsTable.getTable(),
+                        hmsTable.getTableType(), partValues);
+                loadPartitionStats(key);
```

Comment:
I think you could write the value to cache.

Reply:
> I think you could write the value to cache.

ok, now loadPartitionStats(key) output will be put into partitionStatsCache